scrapy_poet/page_input_providers.py

```diff
@@ -1,41 +1,66 @@
-# -*- coding: utf-8 -*-
+"""The Injection Middleware needs a standard way to build dependencies for
```

Comment:
Maybe it can be mentioned here that one use case to implement a provider is to obtain data from external services or APIs like Splash, AutoExtract, etc.

Reply:
Done.